monkey/infection_monkey/monkey.py

```diff
@@ -125,124 +127,36 @@ def start(self):
             if is_running_on_island():
                 WormConfiguration.started_on_island = True
                 ControlClient.report_start_on_island()
-            ControlClient.should_monkey_run(self._opts.vulnerable_port)
+
+            if not ControlClient.should_monkey_run(self._opts.vulnerable_port):
+                raise PlannedShutdownException(
+                    "Monkey shouldn't run on current machine "
+                    "(it will be exploited later with more depth)."
+                )
```

Comment:
Note (in general) this means it won't collect data.
This seems not the right decision

Reply:
I'm not really sure what the intent of this check is. It needs a more descriptive name than "should_monkey_run". This exception will only ever be raised under very specific conditions. 

For this PR, this code preserves the existing functionality, so I'm not overly worried about it. Additionally, it is only raised if the monkey is running on the island as a result of a breach (and some other conditions are also true). Finally, we may do some refactoring in the next release to clean this all up. If that happens, this check may get reworked then.